Formula/molten-vk.rb

```diff
@@ -0,0 +1,137 @@
+class MoltenVk < Formula
+  desc "Implementation of the Vulkan graphics and compute API on top of Metal"
+  homepage "https://github.com/KhronosGroup/MoltenVK"
+  url "https://github.com/KhronosGroup/MoltenVK/archive/v1.0.34.tar.gz"
+  sha256 "a687007c1049fe8277b181c5e403776518e81a8e642af920a135649a058f165b"
+
+  depends_on "cmake" => :build
+  depends_on :xcode => ["10.0", :build]
+  # Requires IOSurface/IOSurfaceRef.h.
+  depends_on :macos => :sierra
+
+  # MoltenVK depends on very specific revisions of its dependencies.
+  # For each resource the path to the file describing the expected
+  # revision is listed.
+  resource "cereal" do
+    # ExternalRevisions/cereal_repo_revision
+    url "https://github.com/USCiLab/cereal.git",
+        :revision => "51cbda5f30e56c801c07fe3d3aba5d7fb9e6cca4"
+  end
+
+  resource "vulkan-headers" do
+    # ExternalRevisions/Vulkan-Headers_repo_revision
+    url "https://github.com/KhronosGroup/Vulkan-Headers.git",
+        :revision => "08cbb5458f692d4778806775f65eb3dc642ddbbf"
+  end
+
+  resource "vulkan-portability" do
+    # ExternalRevisions/Vulkan-Portability_repo_revision
+    url "https://github.com/KhronosGroup/Vulkan-Portability.git",
+        :revision => "53be040f04ce55463d0e5b25fd132f45f003e903"
+  end
+
+  resource "spirv-cross" do
+    # ExternalRevisions/SPIRV-Cross_repo_revision
+    url "https://github.com/KhronosGroup/SPIRV-Cross.git",
+        :revision => "ac5a9570a744eb72725c23c34f36fbc564c0bb51"
+  end
+
+  resource "glslang" do
+    # ExternalRevisions/glslang_repo_revision
+    url "https://github.com/KhronosGroup/glslang.git",
+        :revision => "e06c7e9a515b716c731bda13f507546f107775d1"
+  end
+
+  resource "spirv-tools" do
+    # External/glslang/known_good.json
+    url "https://github.com/KhronosGroup/SPIRV-Tools.git",
+        :revision => "26c1b8878315a7a5c188df45e0bc236bb222b698"
+  end
+
+  resource "spirv-headers" do
+    # External/glslang/known_good.json
+    url "https://github.com/KhronosGroup/SPIRV-Headers.git",
+        :revision => "2434b89345a50c018c84f42a310b0fad4f3fd94f"
+  end
+
+  resource "vulkan-tools" do
+    # ExternalRevisions/Vulkan-Tools_repo_revision
+    url "https://github.com/KhronosGroup/Vulkan-Tools.git",
+        :revision => "2abb69904b9ad017d39d3da1e7fc3dec1a584cd8"
+  end
+
+  def install
+    (buildpath/"External/cereal").install resource("cereal")
```

Comment:
I'd say these can be better installed like this with manual config for the glslang resource and "children".
```
    glslang = resource("glslang")
    (resources - [glslang ]).each do |r|
      r.stage do
        (buildpath/"External/"r).install resource(r)
      end
end
```

Reply:
I'm new to formula here, could you mind expand what is `r.stage` here?
I've seen the resource copied several times when switch to code like this.

Reply:
or something like this?

~~~
    ["Vulkan-Headers", "Vulkan-Portability", "SPIRV-Cross", "glslang", "Vulkan-Tools"].each do |r|
      resource(r).stage(Pathname.pwd/"External"/r)
    end

    resource("glslang_SPIRV-Tools").stage(Pathname.pwd/"External/glslang/External/spirv-tools")
    resource("glslang_SPIRV-Headers").stage(Pathname.pwd/"External/glslang/External/spirv-tools/external/spirv-headers")
~~~